### PR TITLE
ADDED - GET endpoint /users/{userId}/additional-info to fetch additional information from userId

### LINKED ISSUE #215: Feat: Add GET endpoint /users/{userId}/additional-info to fetch additional information from userId
### Is your feature request related to a problem? Please describe.
Although we have endpoint /additional-info for a to fetch additional details of current user (using access_token). We don't have endpoint for a user (org representative) requesting additional-information of other user.

### Describe the solution you'd like
There should be a GET endpoint `/users/{userId}/additional-info` for fetching a user's additional information from userId.

### Additional context
Refer to the discussion [here](https://github.com/anitab-org/bridge-in-tech-web/issues/155).

### PR DESCRIPTION
**Pull request by naveen8801**
### Description

Added GET endpoint /users/{userId}/additional-info to fetch additional information from userId 

Fixes #215 

### Type of Change:

- Code

**Code/Quality Assurance Only**

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Screenshot (59)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by naveen8801** [comment]
Hi @mtreacy002  , can you please explain me the conflict & how to resolve it ?

File: app/api/resources/users.py [hunk 1]
```diff
@@ -513,3 +513,43 @@ def get(cls, user_id):
...
+    @users_ns.response(
+        HTTPStatus.NOT_FOUND.value,
+        f"{messages.ADDITIONAL_INFORMATION_WITH_ID_DOES_NOT_EXIST}",
+    )
+    @users_ns.expect(auth_header_parser, validate=True)
+    def get(cls, user_id):
+        """
+        Returns additional information of user with 'user id' equal to user_id
+
+        A user with valid access token can use this endpoint to view their additional information details.
```
Comment (epicadk) [inline app/api/resources/users.py:574]: Description is misleading. They can view the details of the user with id = userid.

File: config.py [hunk 2]
```diff
@@ -86,9 +86,9 @@ class LocalConfig(BaseConfig):
     DEBUG = True
 
     # Using a local postgre database
-    SQLALCHEMY_DATABASE_URI = "postgresql:///bit_schema"
+    # SQLALCHEMY_DATABASE_URI = "postgresql:///bit_schema"
```
Comment (epicadk) [inline config.py]: Change not required.

File: app/database/models/ms_schema/user.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ class UserModel(db.Model):
     )
 
     def __init__(self, name, username, password, email, terms_and_conditions_checked):
-        """Initialises userModel class with name, username, password, email, and terms_and_conditions_checked. """
```
Comment (epicadk) [inline app/database/models/ms_schema/user.py]: Did black auto format this?

File: app/api/resources/users.py [hunk 4]
```diff
@@ -513,3 +513,43 @@ def get(cls, user_id):
...
+        The endpoint doesn't take any other input.
+        """
+
+        token = request.headers.environ["HTTP_AUTHORIZATION"]
+
+        is_wrong_token = validate_token(token)
+
+        if not is_wrong_token:
```
Comment (epicadk) [inline app/api/resources/users.py]: if is wrong token: 
    return is_wrong_token

File: app/api/resources/users.py [hunk 5]
```diff
@@ -513,3 +513,43 @@ def get(cls, user_id):
...
+            user_json = AUTH_COOKIE["user"].value
+            user = ast.literal_eval(user_json)
```
Comment (epicadk) [inline app/api/resources/users.py]: Where are you using this?

**Review by epicadk — CHANGES_REQUESTED** [review]
Thanks for the pr. Please have a look at my comments .

File: config.py (hunk 2 above)
Comment (naveen8801) [reply config.py]: > Change not required.

Ok @epicadk :)

File: app/api/resources/users.py (hunk 1 above)
Comment (naveen8801) [reply app/api/resources/users.py:574]: > Description is misleading. They can view the details of the user with id = userid.

Ok @epicadk :)

File: app/database/models/ms_schema/user.py (hunk 3 above)
Comment (naveen8801) [reply app/database/models/ms_schema/user.py]: > Did black auto format this?

Yaa , by using command " black . "

File: app/api/resources/users.py (hunk 5 above)
Comment (naveen8801) [reply app/api/resources/users.py]: @epicadk  , removed . thanks 👍

File: app/api/resources/users.py (hunk 4 above)
Comment (naveen8801) [reply app/api/resources/users.py]: @epicadk , should i add another if block ?

Comment (epicadk) [reply app/api/resources/users.py]: Nope just invert the existing block.

**Comment by epicadk** [comment]
Please solve the merge conflicts

**Comment by naveen8801** [comment]
> Please solve the merge conflicts

Hi @mtreacy002  @epicadk  , i am unable to solve this conflict :( , please help !

**Comment by epicadk** [comment]
> > Please solve the merge conflicts
> 
> Hi @mtreacy002 @epicadk , i am unable to solve this conflict :( , please help !

Do `git fetch upstream`
and then `git merge upstream/develop`

**Comment by naveen8801** [comment]
> > > Please solve the merge conflicts
> > 
> > 
> > Hi @mtreacy002 @epicadk , i am unable to solve this conflict :( , please help !
> 
> Do `git fetch upstream`
> and then `git merge upstream/develop`

@epicadk  ,  i have executed both commands but know user.py is showing errors :(

conflict

**Comment by epicadk** [comment]
> > > > Please solve the merge conflicts
> > > 
> > > 
> > > Hi @mtreacy002 @epicadk , i am unable to solve this conflict :( , please help !
> > 
> > 
> > Do `git fetch upstream`
> > and then `git merge upstream/develop`
> 
> @epicadk , i have executed both commands but know user.py is showing errors :(
> 
> conflict

Yeah you'll have to fix these and them run `git merge --continue`

**Comment by vj-codes** [comment]
@naveen8801 your fork is 2 commits behind develop , in case you are having a problem resolving the conflicts as said above just copy-paste users.py from the main repo to your fork and commit.

**Review by mtreacy002 — APPROVED** [review]
Well done for completing the task, @naveen8801 .
I've tested this locally and below is the test summary.

1. Code review: Done
2. Tested all possible outcomes:
    Expected behaviour: 
    a. When target user has not yet provided additional info, it should give 404 response and appropriate message

    b. When target user has provided additional info, it should give 200 response and the target user's additional info

Actual behaviour: as expected
a. 404 response
Screen Shot 2021-05-15 at 5 50 29 pm
 
b. 200 response
Screen Shot 2021-05-15 at 5 54 11 pm

3. Environment:
    OS: MacOS Big Sur
    Browser: MS Edge

Congratulation, @naveen8801 and thank you again for your contribution 🎉  🎉 🎉

**Comment by mtreacy002** [comment]
@epicadk and @vj-codes , can you please help give final review before I merge this to develop? Thanks 😉

**Review by epicadk — APPROVED** [review]

**Review by vj-codes — APPROVED** [review]
LGTM 🎉

**Comment by mtreacy002** [comment]
Thank you @epicadk and @vj-codes ❤️ . Merging it now 😉

**Pull request merged**